### PR TITLE
Clone list/dict on property value get/set

### PR DESCRIPTION
**Pull request by nilsRoettgerAtBB**
## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

# Checklist:

- [x] Pull request title reflects its content
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

# Description:

### CODE REVIEW

**Comment by nilsRoettgerAtBB** [comment]
This PR is a draft because I would like to wait until I know if getPropertyValue will return a clone. Then the example code needs to be improved.

**Comment by nilsRoettgerAtBB** [comment]
@JakaMohorkoDS : Shall I close this PR, because I think it should be yours now?

File: core/corecontainers/src/dictobject_impl.cpp [hunk 1]
```diff
@@ -320,6 +320,32 @@ ErrCode DictImpl::getSerializeId(ConstCharPtr* id) const
...
+
+    auto size = this->hashTable.size();
+    lst->hashTable.reserve(size);
+    for (const auto& item : hashTable)
+    {
+        item.first->addRef();
+        item.second->addRef();
+        lst->hashTable.insert(std::make_pair(item.first, item.second));
+    }
+
```
Comment (dejancrn) [inline core/corecontainers/src/dictobject_impl.cpp:345]: If key or value is itself a mutable object such as dict or list, should it be also cloned? I think yes.

File: core/coreobjects/include/coreobjects/property_object_impl.h [hunk 2]
```diff
@@ -766,6 +767,20 @@ ErrCode GenericPropertyObjectImpl<PropObjInterface, Interfaces...>::setPropertyV
             validatePropertyWrite(prop, valuePtr);
             coerceMinMax(prop, valuePtr);
 
+            const auto ct = prop.getValueType();
+            if (ct == ctList)
+            {
+                ListPtr<IBaseObject> listValue;
+                valuePtr.asPtr<ICloneable<IList>>()->clone(&listValue);
```
Comment (dejancrn) [inline core/coreobjects/include/coreobjects/property_object_impl.h:774]: I like to get in the habit of calling checkErrorInfo on pure interface calls that return ErrCode.

**Review by dejancrn — APPROVED** [review]

**Pull request merged**